### PR TITLE
feat(avo-2586): move update and insert assignment to backend

### PR DESCRIPTION
**Pull request by bertyhell**
https://meemoo.atlassian.net/browse/AVO-2586

twin proxy PR: https://github.com/viaacode/avo2-proxy/pull/639

This saves 623 lines of code that do not need to be shipped to the client and we have more control about the permission checking in the backend.

image

### CODE REVIEW

File: src/assignment/components/DuplicateAssignmentButton.tsx [hunk 1]
```diff
@@ -35,7 +36,13 @@ const DuplicateAssignmentButton: FC<DuplicateAssignmentButtonProps & UserProps>
 			icon={IconName.copy}
 			{...props}
 			onClick={async (e) => {
-				const res = await duplicateAssignment(assignment, user as Avo.User.User);
+				if (!user?.profile?.id) {
+					ToastService.danger(
+						'Je moet ingelogd zijn om een opdracht te kunnen dupliceren'
+					);
```
Comment (SHDBramVandenbussche) [inline src/assignment/components/DuplicateAssignmentButton.tsx:42]: Extract translations

File: src/assignment/views/AssignmentOverview.tsx [hunk 2]
```diff
@@ -358,12 +358,18 @@ const AssignmentOverview: FunctionComponent<AssignmentOverviewProps> = ({
 				break;
 			case 'duplicate':
 				try {
+					if (!user?.profile?.id) {
+						ToastService.danger(
+							'Je moet ingelogd zijn om een opdracht te kunnen dupliceren'
+						);
```
Comment (SHDBramVandenbussche) [inline src/assignment/views/AssignmentOverview.tsx:364]: Extract translations

**Review by SHDBramVandenbussche — APPROVED** [review]

File: src/assignment/components/DuplicateAssignmentButton.tsx (hunk 1 above)
Comment (bertyhell) [reply src/assignment/components/DuplicateAssignmentButton.tsx:42]: fixed in https://github.com/viaacode/avo2-client/pull/1782

File: src/assignment/views/AssignmentOverview.tsx (hunk 2 above)
Comment (bertyhell) [reply src/assignment/views/AssignmentOverview.tsx:364]: fixed in https://github.com/viaacode/avo2-client/pull/1782

**Review by WardVerc1 — APPROVED** [review]

**Pull request merged**